TP53 and telomerase panels for figure 4

sjspielman opened the pull request:
This PR partially addresses Issue #1272  and creates a new script `fig4-tp53-telomerase.R` for populating figure 4 panels.

`fig4-tp53-telomerase.R` produces the following figures:

| Panel | Figure File | 
|-------|-------------|
| A  | `figures/pdfs/fig4/panels/tp53_stranded_roc_panel.pdf` |
| B  | `figures/pdfs/fig4/panels/tp53_scores_by_altered_panel.pdf` |
| C  | `figures/pdfs/fig4/panels/tp53_expression_by_altered_panel.pdf` |
| D | `figures/pdfs/fig4/panels/tp53_scores_boxplot_panel.pdf` and `figures/pdfs/fig4/panels/tp53_scores_boxplot_legend.pdf` |
| E  | `figures/pdfs/fig4/panels/telomerase_scores_boxplot_panel.pdf` |
| F  | _Hazard ratio figure NOT MADE in this PR (see bullets below)_


#### Notable changes:

+ `fig4-tp53-panel.R` has been removed and code has been integrated into `fig4-tp53-telomerase.R`. Those output PDFs (panel D) have also been renamed for clarity since there are more tp53 panels now
+ Code to make the telomerase scores boxplot across cancer groups is commented out from `fig4-telomerase-activities.R`, and the original PDF figure from that script has been removed. 
  + This script will likely be deprecated in a figure PR, and will be replaced with a script for supplementary telomerase activities (remaining tasks from #1272) 
+ Note that the survival analysis hazard ratio forest plot is NOT yet made since that analysis needs to be finalized first in #1276 
  + Do we want to block this on #1276 or move ahead as is?

#### Places for reviewers

+ Not sure yet about dealing with inconsistency between panels _D_ and _E_: both are boxplots of scores (tp53 and telomerase, respectively), but have different styles. Each is ordered by median value of the given variable (so x-axis are in different order), _and_ D highlights mutator points but _E_ colors by cancer group.
One approach could be to also highlight mutators in _E_ so the color schemes match, but the cancer group order will still be different which I think is inevitable. This depends on scientific goals.
+ I removed "Other" from the TP53 violin plots because it is not a robust category for comparison, since it represents unclassified samples. 
+ I added red (for now?) mean+/se to the violin plots to for easier comparison. 

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] The analytical code is documented and contains comments.

jharenza reviewed (commented):
Hi @sjspielman! Thanks for updating these.

I do have a few comments:

>Do we want to block this on https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1276 or move ahead as is?

I think we can move with this one for now since survival is a different module and needs review anyway.

>Not sure yet about dealing with inconsistency between panels D and E: both are boxplots of scores (tp53 and telomerase, respectively), but have different styles. Each is ordered by median value of the given variable (so x-axis are in different order), and D highlights mutator points but E colors by cancer group.
>One approach could be to also highlight mutators in E so the color schemes match, but the cancer group order will still be different which I think is inevitable. This depends on scientific goals.

Since we are now using the smaller number of cancer groups, these plots can be placed side by side, in which case the x-axis order wouldn't matter so much. However, I also had an idea that these can be y-faceted by score, and we can align by TP53 score(?). Aligning them could make it easier to discuss groups which have some relationships (eg HGG, DMG, MB have high scores of both, albeit the correlation plots were not significant with the multiple corrections..). What do you think?

For color, I guess I can go either way. I think I was preferring coloring telomerase by cancer group, but maybe we should see where the hyper and ultra-hypermutant samples lie - maybe there will be an unexpected trend? If we color both in this way, I think it makes sense to facet.

>I removed "Other" from the TP53 violin plots because it is not a robust category for comparison, since it represents unclassified samples.

This category was actually previously designated "wildtype" in other publications, but I did not feel 100% comfortable using "wildtype" in this case. They aren't necessarily unclassified, but they are rather samples which do not have known deleterious _TP53_ alterations, that is, they do not have alterations we would utilize as true positives. So the comparison here is what are the scores of samples with known TP alterations in _TP53_ compared to samples with either wildtype _TP53_ or "Other" alterations (these can be in  `c("TP53", "CDKN2A", "MDM2", "MDM4", "GORAB", "ATM", "ATR", "RB1", "CHEK1", "CHEK2")`) - other _TP53_ pathway genes). We did not get the chance to look at these. We expect this group to have some high scores, but mostly _lower_ scores than those with _TP53_ alterations predicted or known to have functional impact (activation/loss). One of the main premises of this classifier is that having a high score should warrant another look - is there a splice variant in _TP53_, is there another variant which is resulting in LOF of _TP53_? So, I would keep this in, but I am open to calling it something else. Additionally, this classifier has only been tested using somatic data as a positive control, and it is possible that some of the samples in "other" have germline _TP53_ variants (eg - medulloblastomas) that we were unable to openly assess here.

I previously plotted like below ([ref](https://www.cell.com/cell-reports/pdfExtended/S2211-1247(19)31268-9)):
<img width="1009" alt="Screen Shot 2022-04-03 at 11 29 44 AM" src="https://user-images.githubusercontent.com/207034/161435448-1d1798b2-29df-445e-8292-44b6f41112f2.png">

There are a few things to note here: 
1. Not all other _TP53_ pathway genes' alterations give rise to high scores, so it is a bit harder to interpret (C-D)
2. All osteosarcoma samples (E), whose tumors are driven by loss of _TP53_, had high scores, even though we had some samples called "wildtype". This was likely because they had underlying SVs in intron 1 of _TP53_ we could not detect with WXS.

Long explanation, but hopefully that clarifies the motivation for that group, which we can name something else - thoughts on that? We can also delve into what gene alterations those samples with higher scores have, but may not have time.

>I added red (for now?) mean+/se to the violin plots to for easier comparison.

Should this be median +/sd? 


Minor comment: can we hyphenate "ultra-hypermutant" for consistency with the manuscript and previous publication?

sjspielman commented:
> However, I also had an idea that these can be y-faceted by score, and we can align by TP53 score(?).

I'm not sure what you mean "faceted by score." Score is a continuous variable, and I do not see any reason to discretize it.

In terms of the "other" tp53 samples, we can keep them in to show the distribution, but I am _really_ uncomfortable comparing the distributions with the classified activated vs loss categories with a formal stats test, because I still don't see that there is no clear hypothesis about what the "other" samples are. Not having certain mutations (biased by what mutations have been studied already that we have access to), in my mind, is absence of evidence for any hypothesis. @jaclyn-taroni, do you have any thoughts?

> Should this be median +/sd?

If anything else probably should do mean + IQR, since sd tends to "go with" mean and median is more appropriate for a nonparametric. Can update this.

jharenza commented:
> I'm not sure what you mean "faceted by score." Score is a continuous variable, and I do not see any reason to discretize it.

Oh I just meant having the facets on the y axis with the scores still continuous but have the plots paneled by row, like in the sketch below. Top panel being TP53 scores, bottom being EXTEND; and they share the x-axis cancer groups with groups being ordered by TP53 score median as you have them currently. 

 ![image](https://user-images.githubusercontent.com/207034/161580070-710fc79f-6c11-4a74-b1b5-bf7763d97909.jpeg)

sjspielman commented:
I've just pushed some changes:

#### Violins

+ Modularized the code further
+ "Other" is now included in violin plots, but it's _not_ included in the statistical tests. The p-value labels were moved closer to lost/activated groupings to hopefully emphasize this. But I can see how this could be confusing.
+ Because there are SO many "other" points compared to "lost" and "activated" I increased point transparency, but now it's a bit harder to see "activated" points.
+ summary pointrange is now median +/- IQR  


#### Boxplots

+ Code is more modular, and legend is updated to more closely match figure
+ Now as a single figure faceted vertically by scores where x-axis is in tp53 order, and we're using the mutator colors.
  + If we want to use cancer group colors, then this place has to go back to being 2 separate vertical panels. What do we think?
+ Ultrahypermutant ---> Ultra-hypermutant
+ I have "Telomerase score" as a label - do we prefer "Normalized EXTEND scores?"

jharenza commented:
>"Other" is now included in violin plots, but it's not included in the statistical tests. The p-value labels were moved closer to lost/activated groupings to hopefully emphasize this. But I can see how this could be confusing.

Can you add a line between activated/lost showing that the p-value comparison goes to those groups? Maybe we do this in illustrator @jaclyn-taroni ?

>Now as a single figure faceted vertically by scores where x-axis is in tp53 order, and we're using the mutator colors.
If we want to use cancer group colors, then this place has to go back to being 2 separate vertical panels. What do we think?

I like this because you can start to see trends that we saw in the correlation plots - some groups have high TP53 and high telomerase scores, but others (meningioma) have the opposite trend.

>I have "Telomerase score" as a label - do we prefer "Normalized EXTEND scores?"

Commented on #1283 that I think either "Telomerase score" or "Telomerase score (EXTEND)" is good

jaclyn-taroni reviewed (commented):
Overall this is looking good. I am returning a few comments that should get addressed. I am also returning a comment about how to address @jharenza's comment about the annotation with a p-value.

jharenza commented on figures/scripts/fig4-tp53-telomerase-panels.R:
```diff
@@ -0,0 +1,385 @@
+                    ),
+                    color = "firebrick", size = rel(0.6)
+    ) +
+    # Add p-value annotation right by the groups we compare
+    annotate("text", 
+             label = paste0("Wilcoxon P-value = ", pvalue), 
+             x = 1.5, y = pvalue_y, size = 3) +
```
Agreed- didn't know how easy it was in R! 😀

sjspielman commented on figures/scripts/fig4-tp53-telomerase-panels.R:
```diff
@@ -0,0 +1,385 @@
+                   `Ultra-hypermutant` = "red")
+# Other plot parameters which need to be re-introduced in legend:
+normal_alpha <- 0.7
+normal_size <- 1.75
+mutator_size <- 2.25
```
Woops, that was the old legend! Forgot to delete. The legend now lives in `figures/pdfs/fig4/panels/tp53_telomerase_boxplots_panel_legend.pdf` and accompanying plot is in `tp53_telomerase_boxplots_panel.pdf`. From my view, points match up in these two files.

sjspielman commented on figures/scripts/fig4-tp53-telomerase-panels.R:
```diff
@@ -0,0 +1,385 @@
+  labs(x = "Cancer group", 
+       y = "Score") +
+  facet_wrap(~score_type, nrow = 2) +
+  # do we want an hline at 0.5? Might be useful guiding line but also add unnecessary visual noise
+  # geom_hline(yintercept = 0.5) +
```
Deleting comments

sjspielman commented on figures/scripts/fig4-tp53-telomerase-panels.R:
```diff
@@ -0,0 +1,385 @@
+
+
+tp53_expression_plot <- stranded_tp53 %>%
+  rename(tp53 = tp53_expression) %>%
```
done!

sjspielman commented:
Updated with `stat_pvalue_manual()`, removed old legend file, and updated expression violin plots to show `log(fpkm+1)` which is also now reflected in the axis title.

jharenza commented:
P-values look good!

One more thing- can we print out N, R, and p-values (and adjusted p when necessary), or add N to the x-axis in parenthesis for each group, for the respective plots for the manuscript legends? I think this info was previously in notebooks and/or tsv files.

sjspielman commented:
> One more thing- can we print out N, R, and p-values (and adjusted p when necessary), or add N to the x-axis in parenthesis for each group, for the respective plots for the manuscript legends? I think this info was previously in notebooks and/or tsv files.

Which plots are you referring to? We don't have any correlations in these plots. Do you mean adding N to the tp53 violin plots? I can definitely label those x-axes!

jharenza commented:
> > One more thing- can we print out N, R, and p-values (and adjusted p when necessary), or add N to the x-axis in parenthesis for each group, for the respective plots for the manuscript legends? I think this info was previously in notebooks and/or tsv files.
> 
> 
> 
> Which plots are you referring to? We don't have any correlations in these plots. Do you mean adding N to the tp53 violin plots? I can definitely label those x-axes! 

Oh yes was generalizing this with the telomerase TERT/TERC  plots. I'm good whichever way- whether on plots, printed in notebook, or exported table.

jaclyn-taroni reviewed (approved):
The updates look good! I don't think I need to re-review when the sample size gets added to the labels.

sjspielman commented:
> Oh yes was generalizing this with the telomerase TERT/TERC plots.

Ok, these are over in PR #1283 and already have `R` labels. For this PR, I'll add sample sizes to the violin plot x-axis labels.

sjspielman commented:
@jharenza Are the labels I added here what you had in mind? I added `N=` info to x-axes for violin plots and to the mutation status legend.

jharenza commented:
> @jharenza Are the labels I added here what you had in mind? I added `N=` info to x-axes for violin plots and to the mutation status legend.

yes, looks great!
